include/xsimd/types/xsimd_avx512_int32.hpp

```diff
@@ -349,6 +388,43 @@ namespace xsimd
             {
                 return rhs;
             }
+
+            static batch_type sadd(const batch_type& lhs, const batch_type& rhs)
+            {
+                /* origin: /nsimd/include/nsimd/x86/avx512_skylake/adds.h */
+                /*
+                * ====================================================
+                * Copyright (c) 2019 Agenium Scale
+                *
+                * MIT License see https://github.com/agenium-scale/nsimd/blob/master/LICENSE
+                * ====================================================
+                */
+                //todo bench againt unrroled loop
+                //todo factorize int32_t uint32_t
+                const auto ures = _mm512_add_epi32(lhs, rhs);
+                const auto umax = _mm512_set1_epi32(std::numeric_limits<uint32_t>::max());
+                const auto is_overflow = _mm512_cmp_epu32_mask(_mm512_add_epi32(lhs, umax), _mm512_add_epi32(ures, umax), _MM_CMPINT_NLE);
+                return _mm512_mask_blend_epi32(is_overflow, umax, ures);
```

Comment:
I think that this should be `_mm512_mask_blend_epi32(is_overflow, ures, umax);`, according to [this](https://software.intel.com/sites/landingpage/IntrinsicsGuide/#expand=821,282,4952,721,435&text=_mm512_mask_blend_epi32). I agree that this is totally counter-intuitive.